src/components/input/input.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Create Elastic TextArea
+ *
+ * Adds an event listener to a `textarea` elment that responds to input events
+ * by either increasing or decreasing the `rows` attribute based on whether the
+ * `textarea` is scrolling or not. Returns an object containing a `destroy()`
+ * method to remove the event listener.
+ *
+ * 1. Sanity check to prevent an infinite loop if the textarea has a fixed height
+ *    @see https://github.com/cloudfour/cloudfour.com-patterns/pull/716#discussion_r428349375
+ *
+ * @param textarea - the target `textarea` element
+ */
+export const createElasticTextArea = (textarea: HTMLTextAreaElement) => {
+  const maxRows = 500; // 1
+  const minRows = Number(textarea.getAttribute('rows')) || 2;
+  let rows = Number(textarea.getAttribute('rows')) || minRows;
+  let isScrolling = false;
+
+  /** Check if the textarea is currently scrolling */
+  const checkScrolling = () => textarea.scrollHeight > textarea.clientHeight;
+
+  /** Grow until the textarea stops scrolling */
+  const grow = () => {
+    while (isScrolling && rows < maxRows) {
+      rows++;
+      textarea.setAttribute('rows', String(rows));
+      isScrolling = checkScrolling();
+    }
+  };
+
+  /** Shrink until the textarea matches the minimum rows or starts scrolling */
+  const shrink = () => {
+    while (!isScrolling && rows > minRows) {
+      rows--;
+      textarea.setAttribute('rows', String(Math.max(rows, minRows)));
+      isScrolling = checkScrolling();
+
+      if (isScrolling) {
+        grow();
+        break;
+      }
+    }
+  };
+
+  /** Decide whether to grow or shrink the textarea */
+  const update = () => {
+    isScrolling = checkScrolling();
```

Comment:
Feel free to ignore this suggestion if you disagree, but I wondered if we might rename this to something like:
```js
isScrolling = getIsScrolling();
```
At a glance, I gather the following from `getIsScrolling()` without needing to look at the source function:
- `get` => something will be returned
- `getIsScrolling` => I expect `IsScrolling` to be a `boolean`, therefore, `getIsScrolling` must return a `boolean` letting me know if the textarea element is scrolling

Right now, `checkScrolling()` doesn't really help me understand what `checkScrolling()` does. It checks scrolling and...removes event listeners? Calls other functions? Sets the BG color to a different color? It's not very clear to me what `checkScrolling()` might do. 🙂 

Again, feel free to ignore this suggestion if you strongly disagree. I won't be offended. 😉

Reply:
Good call, I'll update.